[RFC] Fix crash when set rightleft on terminal

erw7 opened the pull request:
Fixes #11438.

Backtrace:
```
#0  schar_from_ascii ( p=0x801cc9e112c3 <error: Cannot access memory at address 0x801cc9e112c3>, c=32 ' ') at ../src/nvim/screen.c:5263
#1  0x00007f31460eccc5 in win_line (wp=wp@entry=0x7fffc9df6230, lnum=lnum@entry=11, startrow=startrow@entry=10, endrow=41, nochange=false, number_only=number_only@entry=false) at ../src/nvim/screen.c:4025
#2  0x00007f31460eed8e in win_update (wp=wp@entry=0x7fffc9df6230) at ../src/nvim/screen.c:1403
#3  0x00007f31460f011f in update_screen (type=<optimized out>) at ../src/nvim/screen.c:502
#4  0x00007f3146138ef4 in normal_redraw (s=s@entry=0x7fffd0a5f700) at ../src/nvim/normal.c:1247
#5  0x00007f314613b159 in normal_check (state=0x7fffd0a5f700) at ../src/nvim/normal.c:1324
#6  0x00007f31460accfe in state_enter (s=0x7fffd0a5f700) at ../src/nvim/state.c:28
#7  0x00007f3146143099 in normal_enter (cmdwin=<optimized out>, noexmode=<optimized out>) at ../src/nvim/normal.c:463
#8  0x00007f314618b541 in main (argc=<optimized out>, argv=<optimized out>) at ../src/nvim/main.c:580
```

erw7 committed:
Fix crash when set rightleft on terminal

fixes #11438

justinmk commented on src/nvim/screen.c:
```diff
@@ -4021,10 +4021,16 @@ win_line (
+        int n = 1;
+        if (wp->w_p_rl) {
+          n = -1;
+        }
```
ternary is more readable and less verbose.

```
int n = wp->w_p_rl ? -1 : 1;
```

justinmk commented on test/functional/terminal/buffer_spec.lua:
```diff
@@ -224,6 +225,23 @@ describe(':terminal buffer', function()
+                                                        |
+      {3:-- TERMINAL --}                                    |
+    ]])
+    feed_command('bdelete!')
```
`feed_command` uses `feed()` (AKA `nvim_input`) so it is asynchronous. Better to use `command` usually.
```suggestion
    command('bdelete!')
```

justinmk commented on test/functional/terminal/buffer_spec.lua:
```diff
@@ -224,6 +225,23 @@ describe(':terminal buffer', function()
+    local columns = eval('&columns')
+    feed(string.rep('a', columns))
+    command('set rightleft')
+    wait()
```
did you try without `wait()`? Usually not needed before `screen:expect` .

erw7 replied:
Oops! It is not necessary because it is a remnant during trial and error.

erw7 committed:
Minor fixes

The following corrections will be made as the reviewers point out.

- Change to use ternary operators instead of if statements.
- Remove unnecessary wait().
- Change to use command() instead of feed_command().

blueyed commented:
Cherry-picked into release-0.4: 972dd75
